src/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -268,6 +268,9 @@ public SamReader open(final SamInputResource resource) {
                             bufferedStream.close();
                             bufferedStream = null;
                         }
+
+                        bufferedStream = IOUtil.maybeBufferInputStream(indexMaybe.asUnbufferedInputStream());
```

Comment:
You also need to handle the case where there is no index (ie., `indexMaybe` == null)